rust/rope/src/rope.rs

```diff
@@ -571,33 +572,33 @@ impl Rope {
     ///
     /// Time complexity: technically O(n log n), but the constant factor is so
     /// tiny it is effectively O(n). This iterator does not allocate.
-    pub fn iter_chunks(&self, start: usize, end: usize) -> ChunkIter {
+    pub fn iter_chunks(&self, range: Range<usize>) -> ChunkIter {
         ChunkIter {
-            cursor: Cursor::new(self, start),
-            end,
+            cursor: Cursor::new(self, range.start),
+            end: range.end,
         }
     }
 
     //TODO: implement iter_chunks using ranges and delete this
     pub fn iter_chunks_all(&self) -> ChunkIter {
```

Comment:
as part of this work we can remove this fn, since `iter_chunks(..)` will let us iter all chunks easily.. (see the TODO comment above)